gp-includes/routes/settings.php

```diff
@@ -41,7 +41,7 @@ public function settings_post( $user_id = null ) {
 				$user_id = get_current_user_id();
 			}
 
-			if ( empty( $_POST['_wpnonce'] ) || ! wp_verify_nonce( $_POST['_wpnonce'], 'update-settings_' . $user_id ) ) {
+			if ( ! $this->verify_nonce( 'update-settings_' . $user_id ) ) {
 				gp_notice_set( __( 'An error has occurred. Please try again.', 'glotpress' ), 'error' );
 				$this->redirect( gp_url( '/settings' ) );
 				exit;
```

Comment:
> call without parameters that could then use the class name and current user to create the nonce id

It's not always the user ID. It can be any ID like for a set, project or translation.